### DISCUSSION TITLE
What LIDAR options are available for Create 3 ?

### DISCUSSION THREAD

**Discussion by jerrynsr** [iRobotEducation/create3_docs#43] [Questions]
Hi,

  I just purchased an iRobot Create 3.  Our dog just died, she was 16 1/2 years old.  I plan on coding up a Create 3 using ROS and Python to emulate a dog.  One of the "behaviors" I plan on coding up is for the Create 3 to wake up when a Blink camera detects a motion event, roll to the room closest to the event, and "bark".   What LIDAR do you recommend that won't break the bank and is compatible with ROS and possibly available with a Python 3 library ?

**Comment by shamlian** [comment]
First: I'm sorry to hear about your dog, and I hope our robot brings you some joy!

I have personally used the Create 3 robot with a Raspberry Pi 4 and a SLAMTEC RPLIDAR-A1 using that vendor's ROS 2 package with slam_toolbox, but other hardware and software options exist with ROS 2 drivers.

Please note that I am an iRobot employee, but this post is my own and doesn't necessarily represent iRobot's positions, strategies, or opinions.

**Comment by jerrynsr** [comment]
Thanks.    I'm going to use the USB-C OTG connection scheme for power and connectivity.   The Create 3 doesn't ship until June so I have plenty of time to get the RPI 4 ready and prebuild the ROS 2 environment, get the LIDAR,  and  design a 4" raised  mount for the LIDAR. I/m going to mount a camera too.   I'll probably use a usb speaker for sound output connected to the RPI4, and usb audio in.  I'm not sure if I'm going to use an array or something of my own design.  I'm looking for a compass.  I have a few Betian BN-880 gps+compass in my parts box so maybe I will use that for the compass.
